crypto/conf/conf_def.c

```diff
@@ -360,17 +362,34 @@ static int def_load_bio(CONF *conf, BIO *in, long *line)
                 trim_ws(conf, p);
                 if (!str_copy(conf, psection, &include, p))
                     goto err;
+
+                if (include_dir != NULL) {
+                    size_t newlen = strlen(include_dir) + strlen(include) + 2;
+
+                    include_path = OPENSSL_malloc(newlen);
+                    OPENSSL_strlcpy(include_path, include_dir, newlen);
+                    OPENSSL_strlcat(include_path, "/", newlen);
```

Comment:
Should this "/" append be done only if there is no '/' at the end of include_dir or beginning of include?

Reply:
I copied this pattern from elsewhere.. Would it not work if there was an extra / ?

Reply:
It would work fine. It's just ugly.

Reply:
So my initial thought was that the check would also be a bit ugly - (And potentially need to deal with platform specific path seperators).

Reply:
I see a different issue with this code.  Say we have these two conf files:

/tmp/blah.cnf:
```
.include foo/something.cnf
```

/tmp/foo/something.cnf:
```
.include ../hickup.cnf
```

What hickup.cnf do you expect to be included, /tmp/hickup.cnf or /hickup.cnf?

I compare this to what we have learned to expect from the C preprocessor re relative inclusions

Reply:
(not your fault, @slontis, the code was broken in that regard before you touched it, I know that)

Reply:
You perl docs recommend not using relative paths....
This seems to be a different issue to prepending a path to the existing path..

Reply:
*My* perl docs?
Yes, it's a different issue, it's true...  and I had completely forgotten the caveat in config(5).
So err, ok, relegated to another PR, message received...